packages/frontend/core/src/layouts/workspace-layout.tsx

```diff
@@ -73,8 +75,27 @@ export const WorkspaceLayoutProviders = ({ children }: PropsWithChildren) => {
   const t = useI18n();
   const pushGlobalLoadingEvent = useSetAtom(pushGlobalLoadingEventAtom);
   const resolveGlobalLoadingEvent = useSetAtom(resolveGlobalLoadingEventAtom);
-  const currentWorkspace = useService(WorkspaceService).workspace;
-  const docsList = useService(DocsService).list;
+  const { workspaceService, docsService, editorSettingService } = useServices({
+    WorkspaceService,
+    DocsService,
+    EditorSettingService,
+  });
+  const currentWorkspace = workspaceService.workspace;
+  const docsList = docsService.list;
+  const defaultMode =
+    useLiveData(editorSettingService.editorSetting.settings$)
+      .newDocDefaultMode === 'edgeless'
+      ? DocMode.Edgeless
+      : DocMode.Page;
+
+  useEffect(() => {
+    const dispose = currentWorkspace.docCollection.slots.docCreated.on(id => {
+      docsList.setPrimaryMode(id, defaultMode);
```

Comment:
will it conflict with journal's set?

Reply:
I have handled journal in [this line](https://github.com/toeverything/AFFiNE/pull/8069/files#diff-56d4463fdaec0c5b8898f5f1a174d3b22bd2834f3b116565a60e0056bb9cc0fdR39).

Reply:
move this logic to a separate hook `useAutoDocMode` to reduce component complexity